zebra-state/src/service/arbitrary.rs

```diff
@@ -86,3 +91,56 @@ impl Strategy for PreparedChain {
         })
     }
 }
+
+/// Generate a chain that allows us to make tests for the legacy chain rules.
+///
+/// Arguments:
+/// - `transaction_version_override`: See `LedgerState::height_strategy` for details.
+/// - `transaction_has_valid_network_upgrade`: See `LedgerState::height_strategy` for details.
+/// - `blocks_after_nu_activation`: The number of blocks the strategy will generate
+/// after the provided `network_upgrade`.
+/// - `network_upgrade` - The height at which the strategy will start generating blocks.
+///
+/// Returns:
+/// A generated arbitrary strategy for the provided arguments.
+pub(crate) fn partial_nu5_chain_strategy(
+    transaction_version_override: u32,
+    transaction_has_valid_network_upgrade: bool,
+    blocks_after_nu_activation: u32,
+    // TODO: This argument can be removed and just use Nu5 after we have an activation height #1841
+    network_upgrade: NetworkUpgrade,
+) -> impl Strategy<
+    Value = (
+        Network,
+        Height,
+        zebra_chain::fmt::SummaryDebug<Vec<Arc<Block>>>,
+    ),
+> {
+    (
+        any::<Network>(),
+        NetworkUpgrade::reduced_branch_id_strategy(),
+    )
+        .prop_flat_map(move |(network, random_nu)| {
+            // TODO: update this to Nu5 after we have a height #1841
+            let mut nu = network_upgrade;
+            let nu_activation = nu.activation_height(network).unwrap();
+            let height = Height(nu_activation.0 + blocks_after_nu_activation);
```

Comment:
Doc or code bug: 

The docs say:
> - `network_upgrade` - The height at which the strategy will start generating blocks.

But the code adds `blocks_after_nu_activation` to the network upgrade activation height, then generates `blocks_after_nu_activation` blocks after that height.

Reply:
Would it help to add a test for the strategy, that makes sure the blocks are at the heights you expect?

Reply:
The documentation is wrong for sure, i made an update in https://github.com/ZcashFoundation/zebra/pull/2366/commits/f13bf679c6fa2c7e91066e3fd28541aae286e5f2